engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -1465,11 +1543,17 @@ public void prepareForMigration(VirtualMachineProfile vm, DeployDestination dest
                     long hostId = vm.getVirtualMachine().getHostId();
                     Host host = _hostDao.findById(hostId);
 
-                    volService.grantAccess(volFactory.getVolume(newVol.getId()), host, destPool);
+                    try {
+                        volService.grantAccess(volFactory.getVolume(newVol.getId()), host, destPool);
+                    } catch (Exception e) {
+                        throw new StorageAccessException("Unable to grant access to volume: " + newVol.getId() + " on host: " + host.getId());
+                    }
```

Comment:
Curious, why we need to catch all _Exception_ here? And we are not even logging it then

Reply:
> Curious, why we need to catch all _Exception_ here? And we are not even logging it then

@shwstppr in order to retry the deployment, whenever granting access to a disk fails in any case (chances are storage temporarily inaccessible, failure, etc); and this is being logged in "_VirtualMachineManagerImpl orchestrateStart()_"